### ISSUE TITLE
Is there a gradle plugin?

### ISSUE THREAD

**Issue by blindpirate** [fabric8io/fabric8-maven-plugin#609]
Our team want to use fabric8 but  we are using gradle, so is there a gradle plugin which do the same thing with `fabric8-maven-plugin`？

Anybody knows that?

If not, it means that we have to write it.

Thank you very much!

**Comment by davsclaus** [comment]
There is no gradle plugin. Is it not possible for gradle to call a maven plugin or something?

**Comment by jstrachan** [comment]
I would hope gradle could generate a pom.xml and just delegate to maven for the docker build + fabric8 manifest generation (while leaving gradle to do the build + test?)

**Comment by davsclaus** [comment]
Yeah there is a ton of maven plugins, and for gradle to become succesful it need some kind of coop with maven.

**Comment by RobWin** [comment]
Gradle Gradle Gradle Gradle :)

**Comment by davsclaus** [comment]
Now that Maven has colors, its going to be hard for gradle ;)
... just kidding

I would rather there was some neutral api that worked for both gradle and maven that you can use to build reusable plugins for them (or plugins that can do 90%).

**Comment by RobWin** [comment]
Good idea. A shared lib which can be used by both plugins?

**Comment by rhuss** [comment]
agreed, that it would be awesome to have a build-tool agnostic backend. 

And we are indeed working on that by refactoring internal packages and try to stuff as much as into Maven agnostic modules. (which then in a latter step can be moved out into separate projects, referred to by other builder tools). The introduction of internal 'services' is some result of this effort.

Tbh it's progressing slowly, mostly because of time constraints and that because it is a _huge_ effort (without much visible new features). Also, we found out that we rely on quite some Maven internals, too. But I'm confident that at least we can get the _generator_ (for auto-creating images) and _enricher_ (for enhancing k8s resources) frameworks out of f-m-p. These are the two most important abstractions useful for other build tools, IMO.

So, yes, this is the plan and no, we can't give any ETA ;-)

**Comment by rhuss** [comment]
To see what I mean with _huge_ you might want to have a look at PR #832 which is about refactoring a small part of fmp (i.e. the watch functionality)

**Comment by rohanKanojia** [comment]
Yes, we are trying to refactor all the enrichment and generation parts to get them out of fmp and into [fabric8-kit](https://github.com/fabric8io/fabric8-kit), which can be used by any gradle plugin for build
